File: src/bumperbot_cpp_examples/src/simple_service_server.cpp
#include <rclcpp/rclcpp.hpp>
#include <bumperbot_msgs/srv/add_two_ints.hpp>

using namespace std::placeholders;

class SimpleServiceServer : public rclcpp::Node
{
    private:
    rclcpp::Service<bumperbot_msgs::srv::AddTwoInts>::SharedPtr service_;

public:
    SimpleServiceServer(const std::string& node) : Node(node)
    {
        service_ = create_service<bumperbot_msgs::srv::AddTwoInts>("add_two_ints", std::bind(&SimpleServiceServer::serviceCallback, this, _1, _2));
        RCLCPP_INFO(get_logger(), "Service server is ready.");
    }

    void serviceCallback(std::shared_ptr<bumperbot_msgs::srv::AddTwoInts::Request> req, std::shared_ptr<bumperbot_msgs::srv::AddTwoInts::Response> res)
    {
        RCLCPP_INFO_STREAM(get_logger(), "Request Received: a: " << req->a << " b: " << req->b);
        res->sum = req->a + req->b;
        RCLCPP_INFO_STREAM(get_logger(), "Sending Response: " << res->sum);
    }
    
};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleServiceServer>("simple_service_server");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/bumperbot_py_examples/bumperbot_py_examples/simple_service_server.py
import rclpy
from rclpy.node import Node
from bumperbot_msgs.srv import AddTwoInts


class SimpleServiceServer(Node):
    def __init__(self):
        super().__init__('simple_service_server')

        self.service_ = self.create_service(AddTwoInts, "add_two_ints", self.add_two_ints_callback)

        self.get_logger().info("Simple Service Server has been started.")

    def add_two_ints_callback(self, request, response):
        self.get_logger().info(f"New Request Received a: {request.a}, b: {request.b}")
        response.sum = request.a + request.b
        self.get_logger().info(f"Returning sum: {response.sum}")

        return response

def main(args=None):
    rclpy.init(args=args)
    node = SimpleServiceServer()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()



if __name__ == "__main__":
    main()